### ISSUE TITLE
PerformanceSensitive: No handling of dynamic

### ISSUE THREAD

**Issue by edespong** [dotnet/roslyn-analyzers#2257]
Requested in review of https://github.com/dotnet/roslyn-analyzers/pull/2214#issuecomment-478118996

`dynamic ` is not handled anywhere by the current PerformanceSensitive analyzers other than possibly by accident.

For conversion analyzer for example, this case is will cause 2 boxing operations, but only one is reported.

```
dynamic d3 = DateTime.Today;
long ticks = d3.Ticks;
```

You should obviously not use dynamic in performance sensitive code, so adding a new diagnostic for it would be good as well as handling the cases where other allocations take place.
